x-pack/plugins/enterprise_search/public/applications/shared/not_found/not_found_prompt.test.tsx

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { setMockValues } from '../../__mocks__/kea_logic';
+
+import React from 'react';
+
+import { shallow } from 'enzyme';
+
+import { EuiEmptyPrompt, EuiIcon, EuiButton } from '@elastic/eui';
+
+import { EuiButtonTo } from '../react_router_helpers';
+
+import { NotFoundPrompt } from './';
+
+describe('NotFoundPrompt', () => {
+  const subject = (props?: object) =>
+    shallow(<NotFoundPrompt logo="" productSupportUrl="" {...props} />)
+      .find(EuiEmptyPrompt)
+      .dive();
+
+  it('renders', () => {
+    const wrapper = subject({
+      logo: 'logoAppSearch',
+      productSupportUrl: 'https://discuss.elastic.co/c/enterprise-search/app-search/',
+    });
+
+    expect(wrapper.find('h1').text()).toEqual('404 error');
+    expect(wrapper.find(EuiIcon).prop('type')).toEqual('logoAppSearch');
+    expect(wrapper.find(EuiButtonTo).prop('to')).toEqual('/');
+    expect(wrapper.find(EuiButton).prop('href')).toEqual(expect.stringContaining('//discuss'));
```

Comment:
TIL `stringContaining`

Reply:
Can you explain how this is different than `expect().toContain()`?  We use that a lot in our codebase, this is the first I've seen of this though.

Reply:
Ohh thanks Byron! Both work in this scenario, I just have a terrible spotty memory 🙈 `.toContain()` is shorter here and I totes shoulda used it.

FWIW though I believe the canonical difference is `expect().toContain()` works like `.includes()` - i.e., it works on both arrays and strings.

`expect.stringContaining()` very clearly delineates a string however. I think I used it here because I'd recently used `expect.objectContaining()` and the API there is similar.

So yeah basically 2 different ways of doing similar things, but hey could be worse, we could be working in PHP right?

Reply:
Switched to `.toContain` in https://github.com/elastic/kibana/pull/103198/commits/36b753c4925b05cdd0a420a50d61331eaf569583